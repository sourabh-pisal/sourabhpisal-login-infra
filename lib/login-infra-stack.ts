import { CfnOutput, Stack, StackProps } from "aws-cdk-lib";
import {
  CfnManagedLoginBranding,
  ManagedLoginVersion,
  OAuthScope,
  UserPool,
  UserPoolClient,
  UserPoolDomain,
} from "aws-cdk-lib/aws-cognito";
import { Construct } from "constructs";

export class LoginInfraStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const userPool = new UserPool(this, "user-pool", {
      selfSignUpEnabled: false,
      autoVerify: { email: false, phone: false },
      standardAttributes: {
        email: { required: true, mutable: true },
      },
    });

    const userPoolClient = new UserPoolClient(this, "user-pool-client", {
      userPool,
      authFlows: { user: true, userSrp: true },
      oAuth: {
        flows: { authorizationCodeGrant: true },
        callbackUrls: ["http://localhost:3000"],
        scopes: [OAuthScope.EMAIL, OAuthScope.OPENID],
      },
    });

    const userPoolDomain = new UserPoolDomain(this, "user-pool-domain", {
      userPool,
      cognitoDomain: { domainPrefix: "login-sourabhpisal" },
      managedLoginVersion: ManagedLoginVersion.NEWER_MANAGED_LOGIN,
    });

    new CfnManagedLoginBranding(this, "managed-login-branding", {
      userPoolId: userPool.userPoolId,
      clientId: userPoolClient.userPoolClientId,
      useCognitoProvidedValues: true,
    });

    new CfnOutput(this, "user-pool-id", { value: userPool.userPoolId });
    new CfnOutput(this, "user-pool-client-id", { value: userPoolClient.userPoolClientId });
    new CfnOutput(this, "user-pool-domain-name", { value: userPoolDomain.domainName });
  }
}
