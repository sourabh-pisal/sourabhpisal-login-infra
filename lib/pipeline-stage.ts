import { StackProps, Stage } from "aws-cdk-lib";
import { Construct } from "constructs";
import { LoginInfraStack } from "./login-infra-stack";

export class PipelineStage extends Stage {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
    new LoginInfraStack(this, "SourabhPisalLoginInfraStack", {});
  }
}
