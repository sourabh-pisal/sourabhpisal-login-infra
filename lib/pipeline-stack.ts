import { Stack, StackProps } from "aws-cdk-lib";
import { CodePipeline, CodePipelineSource, ShellStep } from "aws-cdk-lib/pipelines";
import { Construct } from "constructs";
import { PipelineStage } from "./pipeline-stage";

export class PipelineStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const codePipeline = new CodePipeline(this, "pipline", {
      synth: new ShellStep("synth", {
        input: CodePipelineSource.gitHub("sourabh-pisal/sourabhpisal-login-infra", "main"),
        commands: ["npm ci", "npm run build", "npm run cdk synth"],
      }),
    });

    codePipeline.addStage(new PipelineStage(this, "prod", {}));

    codePipeline.buildPipeline();
  }
}
