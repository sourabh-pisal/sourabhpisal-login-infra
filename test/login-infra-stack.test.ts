import { Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { LoginInfraStack } from "../lib/login-infra-stack";

test("matches snapshot", () => {
  const stack = new Stack();
  new LoginInfraStack(stack, "LoginInfraStack", {});
  const template = Template.fromStack(stack);

  expect(template.toJSON()).toMatchSnapshot();
});
