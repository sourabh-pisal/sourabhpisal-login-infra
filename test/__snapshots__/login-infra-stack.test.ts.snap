// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
{
  "Outputs": {
    "userpoolclientid": {
      "Value": {
        "Ref": "userpoolclient6E7D7551",
      },
    },
    "userpooldomainname": {
      "Value": {
        "Ref": "userpooldomain9FDEA85F",
      },
    },
    "userpoolid": {
      "Value": {
        "Ref": "userpool38E431F2",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "managedloginbranding": {
      "Properties": {
        "ClientId": {
          "Ref": "userpoolclient6E7D7551",
        },
        "UseCognitoProvidedValues": true,
        "UserPoolId": {
          "Ref": "userpool38E431F2",
        },
      },
      "Type": "AWS::Cognito::ManagedLoginBranding",
    },
    "userpool38E431F2": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AccountRecoverySetting": {
          "RecoveryMechanisms": [
            {
              "Name": "verified_phone_number",
              "Priority": 1,
            },
            {
              "Name": "verified_email",
              "Priority": 2,
            },
          ],
        },
        "AdminCreateUserConfig": {
          "AllowAdminCreateUserOnly": true,
        },
        "AutoVerifiedAttributes": [],
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "Schema": [
          {
            "Mutable": true,
            "Name": "email",
            "Required": true,
          },
        ],
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "VerificationMessageTemplate": {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
      "UpdateReplacePolicy": "Retain",
    },
    "userpoolclient6E7D7551": {
      "Properties": {
        "AllowedOAuthFlows": [
          "code",
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": [
          "email",
          "openid",
        ],
        "CallbackURLs": [
          "http://localhost:3000",
        ],
        "ExplicitAuthFlows": [
          "ALLOW_USER_SRP_AUTH",
          "ALLOW_USER_AUTH",
          "ALLOW_REFRESH_TOKEN_AUTH",
        ],
        "SupportedIdentityProviders": [
          "COGNITO",
        ],
        "UserPoolId": {
          "Ref": "userpool38E431F2",
        },
      },
      "Type": "AWS::Cognito::UserPoolClient",
    },
    "userpooldomain9FDEA85F": {
      "Properties": {
        "Domain": "login-sourabhpisal",
        "ManagedLoginVersion": 2,
        "UserPoolId": {
          "Ref": "userpool38E431F2",
        },
      },
      "Type": "AWS::Cognito::UserPoolDomain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
